ci/scripts/python_wheel_manylinux_build.sh

```diff
@@ -87,6 +88,8 @@ pushd /tmp/arrow-build
 
 cmake \
     -DARROW_ACERO=${ARROW_ACERO} \
+    -DARROW_AZURE=${ARROW_AZURE} \
+    -DAZURE_SDK_SOURCE=BUNDLED \
```

Comment:
Do we need this?
Can we use vcpkg for Azure SDK for C++ when we build wheels?

Reply:
We can use vcpkg, I did have that working. The problem is we get an old version of the Azure SDK from before managed identity was fixed. 

If we update to the latest version of vcpkg then we can use a vcpkg based installation but that looked like it could have wide reaching implications. How difficult do you think it would be to upgrade vcpkg?

Reply:
https://github.com/apache/arrow/pull/35430 is the last update.
If we upgrade vcpkg, please open a new PR.

Reply:
Could you update vcpkg to avoid using vendored Azure SDK for C++ with vcpkg build?

Reply:
Ok, maybe getting the manylinux build using vcpkg is too ambitious to go in this PR. I would propose keeping `ARROW_AZURE=OFF` in this PR. That way we can get this PR finished faster so that actually implementing the Azure filesystem is unblocked and we can worry about upgrading vcpkg and getting this manylinux build working afterwards.

Reply:
It makes sense.
The approach will keep each PR as smaller as possible. Let's use the approach.